stock_quant_merge/models/stock.py

```diff
@@ -0,0 +1,49 @@
+# -*- coding: utf-8 -*-
+# © 2015 OdooMRP team
+# © 2015 AvanzOSC
+# © 2015 Serv. Tecnol. Avanzados - Pedro M. Baeza
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from openerp import models, api
+
+
+class StockQuant(models.Model):
+    _inherit = 'stock.quant'
+
+    @api.multi
+    def _mergeable_domain(self):
+        """Return the quants which may be merged with the current record"""
+        self.ensure_one()
+        return [('id', '!=', self.id),
+                ('product_id', '=', self.product_id.id),
+                ('lot_id', '=', self.lot_id.id),
+                ('package_id', '=', self.package_id.id),
+                ('location_id', '=', self.location_id.id),
+                ('reservation_id', '=', False),
+                ('propagated_from_id', '=', self.propagated_from_id.id)]
+
+    @api.multi
+    def merge_stock_quants(self):
+        # Get a copy of the recorset
+        pending_quants = self.browse(self.ids)
```

Comment:
For the main part I think it's OK because the module itself only calls this method on unreserved quants, but there may be corner cases if called through RPC or custom modules.

I'm sorry but it's a bit too subtle for me to research given the time I can spare on this.
Maybe you could propose a test to make sure it's OK?